npg_qc_viewer/t/client/test_uqc_outcomes_view.js

```diff
@@ -0,0 +1,217 @@
+/* globals $, QUnit, requirejs */
+"use strict";
+requirejs.config({
+  baseUrl: '../../root/static',
+  paths: {
+    'qunit': 'bower_components/qunit/qunit/qunit',
+    jquery:  'bower_components/jquery/dist/jquery'
+  }
+});
+
+requirejs([
+  'scripts/qcoutcomes/qc_outcomes_view',
+  'scripts/qcoutcomes/qc_utils'
+], function(qc_outcomes,
+            qc_utils
+) {  
+
+    QUnit.test('Parsing RPT keys', function (assert) {
+      var rptKeys = qc_outcomes._parseRptKeys('results_summary');
+      var expected = ['18245:1', '18245:1:1', '18245:1:2','19001:1', '19001:1:1', '19001:1:2'];
+      assert.deepEqual(rptKeys, expected, 'Correct rpt keys');
+    });
+
+    QUnit.test("Parameter validation tests for utilityDisplaySwitch", function (assert) {
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch();
+        },
+        /Parameter obj must be defined and of type \(DOM\) Element/i,
+        'Throws with no params'
+      );
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(1);
+        },
+        /Parameter obj must be defined and of type \(DOM\) Element/i,
+        'Throws with one (non DOM) param'
+      );
+      var some_element = $('td.lane')[0];
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(some_element);
+        },
+        /utility should be a string/i,
+        'Throws with one param'
+      );
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(some_element, true);
+        },
+        /utility should be a string/i,
+        'Throws with non string parameter'
+      );
+    });
+
+    QUnit.test('Updating for a uqc element with undefined uqc outcome', function(assert) {
```

Comment:
There is no code for update, so there cannot be any test either. Seems wrong test description